src/components/TemplateList/TemplateItem.tsx

```diff
@@ -0,0 +1,110 @@
+import { ITemplateDataDb } from "@/interfaces/template";
+import { Box, Button, Flex, Text } from "@chakra-ui/react";
+import { useState } from "react";
+import CourtTags from "./CourtTags";
+import TemplateDetail from "./TemplateDetail";
+import Image from "next/image";
+import { startSelectTemplate } from "@/store/reducer/buttonToggleSlice";
+import { useStoreSelector } from "@/store/hooks";
+import { useDispatch } from "react-redux";
+import { FaEllipsisH } from "react-icons/fa";
+
+interface Props {
+  template: Omit<ITemplateDataDb, "__v" | "isDeleted">;
+}
+
+const TemplateItem = (prop: Props) => {
+  const dispatch = useDispatch();
+  const { isTemplateSelect } = useStoreSelector((state) => state.buttonToggle);
+  const [hoverOn, setHoverOn] = useState<boolean>(false);
+  const [enableTempDetail, setEnableTempDetail] = useState<boolean>(false);
+  const highLightItem = isTemplateSelect && !hoverOn ? false : true;
```

Comment:
what is the _isTemplateSelect_ purpose?

Reply:
When the user starts to select or hover over the template, other templates' opacity will be reduced. The isTemplateSelect state is to listen to the hover or click, otherwise, all templates' opacity is 0.4 in the beginning.